File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/interfaces/BaseAttributes.java
package com.emerchantpay.gateway.genesisandroid.api.interfaces;

import com.emerchantpay.gateway.genesisandroid.api.util.RequestBuilder;

public interface BaseAttributes {

    RequestBuilder requestBuilder = new RequestBuilder("");

    // Base Params
    default BaseAttributes setTransactionId(String transactionId) {
        requestBuilder.addElement("transaction_id", transactionId);
        return this;
    }

    default BaseAttributes setUsage(String usage) {
        requestBuilder.addElement("usage", usage);
        return this;
    }

    default BaseAttributes setRemoteIp(String remoteIP) {
        requestBuilder.addElement("remote_ip", remoteIP);
        return this;
    }

    default RequestBuilder buildBaseParams() {
        return requestBuilder;
    }
}

File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/interfaces/BaseAttributes.kt
package com.emerchantpay.gateway.genesisandroid.api.interfaces

import com.emerchantpay.gateway.genesisandroid.api.util.RequestBuilder

interface BaseAttributes {

    // Base Params
    fun setTransactionId(transactionId: String): BaseAttributes {
        requestBuilder.addElement("transaction_id", transactionId)
        return this
    }

    fun setUsage(usage: String): BaseAttributes {
        requestBuilder.addElement("usage", usage)
        return this
    }

    fun setRemoteIp(remoteIP: String): BaseAttributes {
        requestBuilder.addElement("remote_ip", remoteIP)
        return this
    }

    fun buildBaseParams(): RequestBuilder {
        return requestBuilder
    }

    companion object {

        val requestBuilder = RequestBuilder("")
    }
}